### PR TITLE
feat: added schedule date

### PR DESCRIPTION
**Pull request by katungi**
## Description
I added a  nullable field in the user records and a custom route to allow user records to be updated.

## How Has This Been Tested?
I have performed tests using postman

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added necessary inline code documentation
- [x] I have added tests that prove my fix is effective and that this feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by katungi — COMMENTED** [review]
@wathigo I am not sure about what i have done and would definitely like to hear from you.

File: app/controllers/records_controller.rb [hunk 1]
```diff
@@ -27,10 +27,14 @@ def destroy
     head :no_content
   end
 
+  def update_records
+    @record_update = Record.find(params[:id])
+    @record_update.update(schedule_date: params[:schedule_date])
+  end
```
Comment (wathigo) [inline app/controllers/records_controller.rb]: Let's also update nullable `specialist_id`. This is where a specialist claims a user record. We need to update both `shedule_date` and `specialist_id`

**Review by wathigo — CHANGES_REQUESTED** [review]
Great work Daniel!
Check the comment :point_down:

File: app/controllers/records_controller.rb (hunk 1 above)
Comment (katungi) [reply app/controllers/records_controller.rb]: Let me work on it asap. But I assume the route looks okay?? that is where i had my doubts
@wathigo

Comment (katungi) [reply app/controllers/records_controller.rb]: @wathigo check it out now

Comment (wathigo) [reply app/controllers/records_controller.rb]: Let me see

File: config/routes.rb [hunk 2]
```diff
@@ -3,12 +3,14 @@
...
   post 'auth/login', to: 'authentication#authenticate'
 
-  post 'signup', to: 'users#create'
+  post 'auth/signup', to: 'users#create'
 
   resources :users, only: %i[show update destroy] do
     resources :records
   end
 
+  put '/users/:user_id/records/:id', to: 'records#update_records'
```
Comment (wathigo) [inline config/routes.rb]: Let's use a custom name for the route. We can get the user_id from the record, `specialist_id` from the `current_user`. We just need to pass `record_id` and `shedule_date` as parameters in our request payload on the front-end.
So we can have:
```ruby
put 'claim_user_record', to: 'records#update_records'.
```

**Review by wathigo — CHANGES_REQUESTED** [review]
Hi Daniel check the comment above :point_up:

File: config/routes.rb (hunk 2 above)
Comment (katungi) [reply config/routes.rb]: That makes alot of sense.

**Review by wathigo — APPROVED** [review]
LGTM!

**Review by Danielshow — APPROVED** [review]
Lgtm

**Pull request merged**